pkg/component/watch.go

```diff
@@ -120,7 +139,11 @@ func WatchAndPush(client *occlient.Client, componentName string, applicationName
 						break
 					}
 				}
-				if !alreadyInChangedFiles {
+				matched, err := isRegExpMatch(event.Name, ignores)
```

Comment:
Why are we checking for ignored files here?
If the file was ignored, the watch shouldn't be added to it and there shouldn't be event for the ignored file.

Reply:
@kadel Because, the watch on a directory watches every file and folder immediately(not recursive though) under it and notifies our code of the changes in them. If we remove the watch on parent directory and only watch end/leaf level files then we run the risk of not detecting new files created(which necessiates us to watch parent directories). I have noticed even with the fixes in this PR when I follow the steps under on windows, the issue is still reproducible:

1. Start watch on local source(deployed as a component)
2. Create a new file in any of the not-ignored folders

And to my suprise I just replaced maketar in our code with maketar in https://github.com/kubernetes/kubernetes/blame/master/pkg/kubectl/cmd/cp.go#L312-L320 and I didn't see the above issue even on windows for most of the watch scenarios(of local source) that I tested, none of them failed...